python/sglang/srt/managers/scheduler.py

```diff
@@ -916,6 +916,10 @@ def _extend_requests_to_queue(self, reqs: List[Req], is_retracted: bool = False)
         if self.disaggregation_mode == DisaggregationMode.DECODE:
             self.disagg_decode_prealloc_queue.extend(reqs)
         else:
+            now = time.time()
+            for req in reqs:
+                if req.queue_time_start is None:
+                    req.queue_time_start = now
```

Comment:
can we move it after `if add_to_grammar_queue:`?

```
        if add_to_grammar_queue:
            req.queue_time_start = time.time()
            self.grammar_queue.append(req)
```

Reply:
yes. Either way works. I moved it to `if add_to_grammar_queue`.